Workbooks/Intune Devices/Devices/IntuneDevices.workbook

```diff
@@ -0,0 +1,351 @@
+            "isRequired": true,
+            "value": {
+              "durationMs": 2419200000
+            },
+            "typeSettings": {
+              "selectableValues": [
+                {
+                  "durationMs": 259200000
+                },
+                {
+                  "durationMs": 604800000
+                },
+                {
+                  "durationMs": 1209600000
+                },
+                {
+                  "durationMs": 2419200000
+                },
+                {
+                  "durationMs": 2592000000
+                },
+                {
+                  "durationMs": 5184000000
+                },
+                {
+                  "durationMs": 7776000000
+                }
+              ],
+              "allowCustom": true
+            },
+            "timeContext": {
+              "durationMs": 86400000
+            },
+            "label": "Time Range"
+          }
+        ],
+        "style": "pills",
+        "queryType": 0,
+        "resourceType": "microsoft.operationalinsights/workspaces"
+      },
+      "name": "Parameters"
+    },
+    {
+      "type": 1,
+      "content": {
+        "json": "## Number Of Devices Trends\r\n---"
+      },
+      "name": "text - NumberOfDevicesTrends"
+    },
+    {
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "IntuneDevices\r\n| where Result != \"Completed\"\r\n| summarize NumberOfDevices = dcount(DeviceId) by BatchId\r\n| join (\r\nIntuneDevices\r\n| where Result == \"Completed\"\r\n| project BatchTime = TimeGenerated, BatchId\r\n| summarize arg_max(BatchId, *) by BatchId\r\n) on BatchId\r\n| project BatchTime, NumberOfDevices",
+        "size": 3,
+        "aggregation": 5,
+        "showAnnotations": true,
+        "timeContext": {
+          "durationMs": 0
+        },
+        "timeContextFromParameter": "TimeRange",
+        "queryType": 0,
+        "resourceType": "microsoft.operationalinsights/workspaces",
+        "visualization": "timechart"
+      },
+      "name": "Query - NumOfDevicesTimeChart"
+    },
+    {
+      "type": 1,
+      "content": {
+        "json": "## Devices By OS\r\n---"
+      },
+      "customWidth": "50",
+      "name": "Text - DevicesByOS"
+    },
+    {
+      "type": 1,
+      "content": {
+        "json": "## Devices By Ownership\r\n---"
+      },
+      "customWidth": "50",
+      "name": "Text - DevicesByOwnership"
+    },
+    {
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "IntuneDevices\r\n| where TimeGenerated > ago(1d)\r\n| where Result != \"Completed\"\r\n| join kind = inner (\r\n    IntuneDevices\r\n    | where TimeGenerated > ago(1d)\r\n    | where Result == \"Completed\"\r\n    | summarize arg_max(TimeGenerated, BatchId)\r\n) on BatchId\r\n| summarize arg_max(DeviceId, *) by DeviceId\r\n| summarize count() by OS",
```

Comment:
you have a time range parameter but don't appear to be using it?  it looks like the queries are all hardcoded to ago(1d)?

it looks like you could just search/replace `where TimeGenerated > ago(1d)` with `where TimeGenerated {TimeRange}` ?